app/services/dashboard_sequence_runlist.rb

```diff
@@ -80,14 +80,18 @@ def run_answers(run, submissions_created_after)
     if num_tries < 1
       return []
     end
-    submission = submissions.last
-    return {
-      page: submission.interactive_page_id,
-      pageIndex: submission.interactive_page.position,
-      tryCount: num_tries,
-      numQuestions: 4,
-      answers: submission_answers(submission)
-    }
+    submissions = submissions.group_by(&:interactive_page_id)
+    submissions.map do |page_id, sumbissions|
```

Comment:
It is confusing that the loop variable is the same as the variable being looped over.
Perhaps the loop variable could be changed to `page_submissions`